model.py

```diff
@@ -3695,6 +3695,12 @@ def _restrict_to_customlist_subquery_condition(
         qu = qu.filter(condition)
         return qu
 
+    def complaints_index(self):
```

Comment:
I'm not totally opposed to putting this in the model, but it seems like it might fit better inside a controller-type class?